app/controllers/oaisys/pmh_controller.rb

```diff
@@ -17,59 +17,71 @@ class Oaisys::PMHController < Oaisys::ApplicationController
   def bad_verb; end
 
   def identify
-    expect_no_args for_verb: :Identify
-
     respond_to do |format|
       format.xml { render :identify }
     end
   end
 
   def list_sets
-    expect_args for_verb: :ListSets, exclusive: [:resumptionToken]
+    expect_args exclusive: [:resumptionToken]
   end
 
   # TODO: Handle the identifier argument.
   def list_metadata_formats
-    expect_args for_verb: :ListMetadataFormats, optional: [:identifier]
+    expect_args optional: [:identifier]
 
     respond_to do |format|
       format.xml do
-        render :list_metadata_formats, locals: { supported_formats: SUPPORTED_FORMATS }
+        render :list_metadata_formats, locals: { supported_formats: SUPPORTED_FORMATS, parameters: @parameters }
       end
     end
   end
 
   def list_records
-    expect_args for_verb: :ListRecords, required: [:metadataPrefix], optional: [:from, :until, :set],
-                exclusive: [:resumptionToken]
+    expect_args required: [:metadataPrefix], optional: [:from, :until, :set], exclusive: [:resumptionToken]
   end
 
   # get_record is referring to the verb, not a getter.
   # rubocop:disable Naming/AccessorMethodName
   def get_record
-    expect_args for_verb: :GetRecord, required: [:identifier, :metadataPrefix]
+    expect_args required: [:identifier, :metadataPrefix]
   end
   # rubocop:enable Naming/AccessorMethodName
 
+  # TODO: Handle from, until, and resumptionToken arguments.
   def list_identifiers
-    expect_args for_verb: :ListIdentifiers, required: [:metadataPrefix], optional: [:from, :until, :set],
-                exclusive: [:resumptionToken]
+    expect_args required: [:metadataPrefix], optional: [:from, :until, :set], exclusive: [:resumptionToken]
+
+    metadata_prefix = params[:metadataPrefix]
+    if metadata_prefix == 'oai_dc'
+      results = Oaisys::Engine.config.oai_dc_model.public_items
+    elsif metadata_prefix == 'oai_etdms'
+      results = Oaisys::Engine.config.oai_etdms_model.public_items
+    else
+      raise Oaisys::CannotDisseminateFormatError.new(parameters: @parameters.slice(:verb, :metadataPrefix))
+    end
+    results = results.belongs_to_set(params[:set].tr(':', '/')) if params[:set].present?
```

Comment:
I'm not sure `tr` is the method we want here (I'll be honest, I'd never heard of it) based on its somewhat odd padding characteristics: 
```Returns a copy of str with the characters in from_str replaced by the corresponding characters in to_str. If to_str is shorter than from_str, it is padded with its last character in order to maintain the correspondence.```

`gsub` is probably the more natural call here?

Reply:
Rubocop gives me this:

C: Performance/StringReplacement: Use tr instead of gsub. (https://github.com/JuanitoFatas/fast-ruby#stringgsub-vs-stringtr-code)
    .gsub()

Which is how I discovered tr in the first place, should I still use gsub and ignore the rule in .rubocop.yml?

Reply:
Rubocop gives me this when using gsub (I'm assuming it checks that I'm replacing a single character with a single character):

C: Performance/StringReplacement: Use tr instead of gsub. (https://github.com/JuanitoFatas/fast-ruby#stringgsub-vs-stringtr-code)
    ....gsub(':', '/')...

Which is how I discovered tr in the first place, should I still use gsub and add a rule to ignore this in .rubocop.yml?

Reply:
Rubocop gives me this when using gsub (I'm assuming it checks that I'm replacing a single character with a single character):

C: Performance/StringReplacement: Use tr instead of gsub. (https://github.com/JuanitoFatas/fast-ruby#stringgsub-vs-stringtr-code)
    ....gsub(':', '/')...

Which is how I discovered tr in the first place, should I still use gsub and add a rule to ignore this in .rubocop.yml?

Reply:
Huh, that's interesting. I like the performance gain (it makes sense in that tr doesn't have to compile a regex) but the padding behaviour just strikes me as a mysterious bug waiting to happen....

I'm cool if you want to leave it as-is. I'll learn to love `tr`